src/AppRoot.tsx

```diff
@@ -254,19 +270,51 @@ class AppRoot extends React.Component<RouteComponentProps, State> {
     checkCardInterval = 0 // To indicate setInterval is not running.
   }
 
+  public storeLongValueOnCard = async (longValueBase64: string) => {
+    const formData = new FormData()
+    formData.append('long_value', longValueBase64)
+
+    await fetch('/card/write_long_b64', {
+      method: 'post',
+      body: formData,
+    })
+  }
+
+  public clearLongValueOnCard = async () => {
+    await this.storeLongValueOnCard('')
+  }
+
+  public startCardWritePolling = () => {
+    if (cardWriteInterval === 0) {
+      cardWriteInterval = window.setInterval(async () => {
+        if (
+          lastVoteSaveToCard < lastVoteUpdate &&
+          lastVoteUpdate < Date.now() - GLOBALS.CARD_WRITE_DELAY
+        ) {
+          lastVoteSaveToCard = Date.now()
+
+          const longValue = btoa(JSON.stringify(this.state.votes))
+
+          await this.storeLongValueOnCard(longValue)
```

Comment:
Should we only call this if `longValue` is different from the last time?

Reply:
In general, how do we want the server to know which write wins? In the event of requests `A` and `B` that are handled in the order `B` then `A`, how do we make sure `A` doesn't override `B`?

Reply:
One possible solution: don't use `setInterval`, instead use `setTimeout` and only schedule a new timeout once the request has completed (or failed).

Reply:
@eventualbuddha good call on being extra safe about order of operations here. Will address.